File: Bug_Hunt_Practice/Question 1/solution.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

// Helper function to check for vowels
bool isVowel(char c) {
    return (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u');
}

int calculateTeamPotential(string candidates, int ratings[], int size) {
    int teamPotentialScore = 0;
    for (int i = 0; i < size; i++) {
        char currentCandidate = candidates[i];
        int currentRating = ratings[i];
    
        if (currentRating > 5 || isVowel(currentCandidate)) {
            teamPotentialScore += currentRating;
        }
    }
    return teamPotentialScore;
}

int main() {
    string candidates = "axbyciz";
    int ratings[] = {8, 10, 3, 6, 9, 2, 7};
    int size = 7;
    
    int score = calculateTeamPotential(candidates, ratings, size);
    cout << "Expected Output: 8" << endl;
    cout << "Actual Output:   " << score << endl;
    return 0;
}

File: Bug_Hunt_Practice/Question 1/solution.py
def is_vowel(c):
    """Helper function to check for vowels"""
    return c in 'aeiou'

def calculate_team_potential(candidates, ratings):
    team_potential_score = 0
    for i in range(len(candidates)):
        current_candidate = candidates[i]
        current_rating = ratings[i]
        
        if current_rating > 5 or is_vowel(current_candidate):
            team_potential_score += current_rating
            
    return team_potential_score

# --- Main execution ---
candidates = "axbyciz"
ratings = [8, 10, 3, 6, 9, 2, 7]

score = calculate_team_potential(candidates, ratings)

print(f"Expected Output: 8")
print(f"Actual Output:   {score}")